core/src/test/java/org/apache/iceberg/TestTables.java

```diff
@@ -264,8 +265,17 @@ public void commit(TableMetadata base, TableMetadata updatedMetadata) {
             throw new CommitFailedException("Injected failure");
           }
           Integer version = VERSIONS.get(tableName);
+          String metadataLocation =
```

Comment:
why are these changes necessary?

Reply:
the method `lazyTable` in `SerializableTable` will load metadata file,  while the `TestTable` has not write metadata file

Reply:
I still think those changes are not necessary. The test passed fine for me without having those changes